ingest/ingest_pipeline.py

```diff
@@ -519,44 +527,64 @@ def run_ingest(ingest, arguments, parsed_args):
     """Logs any errors, then exits Ingest Pipeline with standard OS code
     """
     if len(status) > 0:
-        if all(i < 1 for i in status):
+        # for successful DE jobs, need to delocalize results
+        if "differential_expression" in arguments and all(i < 1 for i in status):
+            file_path, study_file_id = get_delocalization_info(arguments)
+            # append status?
+            if IngestFiles.is_remote_file(file_path):
+                files_to_match = DifferentialExpression.string_for_output_match(
+                    arguments
+                )
+                DifferentialExpression.delocalize_de_files(
+                    file_path, study_file_id, files_to_match
+                )
+        # all non-DE ingest jobs can exit on success
+        elif all(i < 1 for i in status):
             sys.exit(os.EX_OK)
         else:
-            # delocalize errors file
-            for argument in list(arguments.keys()):
-                captured_argument = re.match("(\w*file)$", argument)
-                if captured_argument is not None:
-                    study_file_id = arguments["study_file_id"]
-                    matched_argument = captured_argument.groups()[0]
-                    file_path = arguments[matched_argument]
-                    if IngestFiles.is_remote_file(file_path):
-                        # Delocalize support log
-                        IngestFiles.delocalize_file(
-                            study_file_id,
-                            arguments["study_id"],
-                            file_path,
-                            "log.txt",
-                            f"parse_logs/{study_file_id}/log.txt",
-                        )
-                        # Delocalize user log
-                        IngestFiles.delocalize_file(
-                            study_file_id,
-                            arguments["study_id"],
-                            file_path,
-                            "user_log.txt",
-                            f"parse_logs/{study_file_id}/user_log.txt",
-                        )
-                    # Need 1 argument that has a path to identify google bucket
-                    # Break after first argument
-                    break
+            file_path, study_file_id = get_delocalization_info(arguments)
+            if IngestFiles.is_remote_file(file_path):
+                # Delocalize support log
+                IngestFiles.delocalize_file(
+                    study_file_id,
+                    arguments["study_id"],
+                    file_path,
+                    "log.txt",
+                    f"parse_logs/{study_file_id}/log.txt",
+                )
+                # Delocalize user log
+                IngestFiles.delocalize_file(
+                    study_file_id,
+                    arguments["study_id"],
+                    file_path,
+                    "user_log.txt",
+                    f"parse_logs/{study_file_id}/user_log.txt",
```

Comment:
Putting DE log output into a DE-specific folder might help debuggability, by distinguishing DE / non-DE at a glance.

```suggestion
                    f"parse_logs/differential_expression/{study_file_id}/log.txt",
                )
                # Delocalize user log
                IngestFiles.delocalize_file(
                    study_file_id,
                    arguments["study_id"],
                    file_path,
                    "user_log.txt",
                    f"parse_logs/differential_expression/{study_file_id}/user_log.txt",
```

Reply:
`b709b84`